### PR TITLE
end_user/feature/all_cuisines

### PR DESCRIPTION
**Pull request by mosayed01**
# Tasks
- [x] add all cuisines screen from API
- [x] Navigate from home to all cuisines screen
- [x] Refactor preferred screen to get cuisines from API
- [x] Add root extension val on Voyager Navigator to navigate from root parent
- [x] Make accepted language from local gateway in an interceptor
- [x] move app bar to home screen
- [x] fix composition local issue because no theme wrapper in direct composable app we call .Content inside 
- [x] add dynamic koin upload module; then remove it
- [x] fix network issue
# preview
[all cuisines.webm](https://github.com/TheChance101/beep-beep/assets/83292287/b958cbae-e701-4e07-9c36-aa4303c0a623)

### CODE REVIEW

File: client_end_user/shared/src/commonMain/kotlin/presentation/composable/PreferredFoodCard.kt [hunk 1]
```diff
@@ -20,43 +20,42 @@ import androidx.compose.ui.draw.clip
...
+    ) {
         Text(
-            text = state.name,
-            color= Theme.colors.contentSecondary,
+            text = state.cuisineName,
+            color = Theme.colors.contentSecondary,
             style = Theme.typography.title,
             modifier = Modifier.padding(start = 12.dp, top = 12.dp, bottom = 4.dp)
         )
         Image(
```
Comment (ahmednasserzaza) [inline client_end_user/shared/src/commonMain/kotlin/presentation/composable/PreferredFoodCard.kt:53]: Shall we use ImagePicker in composables we made or is not neccessary here !

**Review by ahmednasserzaza — APPROVED** [review]

**Review by HaidyAbuGom3a — APPROVED** [review]

**Review by ay0o5h — APPROVED** [review]

File: client_end_user/shared/src/commonMain/kotlin/di/screenModelsModule.kt [hunk 2]
```diff
@@ -27,5 +26,6 @@ val screenModelsModule = module {
     factoryOf(::PickLanguageScreenModel)
     factoryOf(::PreferredRideScreenModel)
     factoryOf(::NotificationScreenModel)
+    factoryOf(::CuisinesScreenModel)
     factory { AppScreenModel(get()) }
```
Comment (AzizaHelmy) [inline client_end_user/shared/src/commonMain/kotlin/di/screenModelsModule.kt:30]: why this model only use this approach in injection?, i think it missed, so we need to refactor it

File: client_end_user/shared/src/commonMain/kotlin/presentation/home/HomeScreen.kt [hunk 3]
```diff
@@ -325,4 +356,23 @@ class HomeScreen :
             }
         }
     }
+
+    @Composable
+    private fun Wallet(
+        modifier: Modifier = Modifier,
+        value: String
```
Comment (AzizaHelmy) [inline client_end_user/shared/src/commonMain/kotlin/presentation/home/HomeScreen.kt:363]: arrange the order of these attributes

File: client_end_user/shared/src/commonMain/kotlin/presentation/home/composable/CuisineCard.kt [hunk 4]
```diff
@@ -23,38 +24,43 @@ import androidx.compose.ui.Alignment
...
+import presentation.cuisines.CuisineUiState
 import resources.Resources
 
 @Composable
 fun CuisineCard(
     cuisine: CuisineUiState,
     onClickCuisine: (cuisineId: String) -> Unit,
     modifier: Modifier = Modifier,
+    width: Dp = 76.dp,
+    backGround: Color = Theme.colors.secondary,
```
Comment (AzizaHelmy) [inline client_end_user/shared/src/commonMain/kotlin/presentation/home/composable/CuisineCard.kt:43]: backGround is a one word, replace it with "backgroundColor"

File: client_end_user/shared/src/commonMain/kotlin/resources/strings/IStringResources.kt [hunk 5]
```diff
@@ -25,6 +25,7 @@ interface IStringResources {
     val notification: String
```
Comment (AzizaHelmy) [inline client_end_user/shared/src/commonMain/kotlin/resources/strings/IStringResources.kt:25]: i just need to know why this interface for StringResource?
is this for Localiztion or what?

**Review by AzizaHelmy — APPROVED** [review]
i didn't find any valuable comments to add as a review to this code, ingeneral great job 👏

File: client_end_user/shared/src/commonMain/kotlin/resources/strings/IStringResources.kt (hunk 5 above)
Comment (mosayed01) [reply client_end_user/shared/src/commonMain/kotlin/resources/strings/IStringResources.kt:25]: Yup it's using for localization

File: client_end_user/shared/src/commonMain/kotlin/di/screenModelsModule.kt (hunk 2 above)
Comment (mosayed01) [reply client_end_user/shared/src/commonMain/kotlin/di/screenModelsModule.kt:30]: Sadly it's out of PR scope.. may we chance it in another PR

**Pull request merged**